src/algebra/homology/homotopy.lean

```diff
@@ -270,6 +282,142 @@ def comp_right_id {f : C ⟶ C} (h : homotopy f (𝟙 C)) (g : C ⟶ D) : homoto
+  homotopy (null_homotopic_map hom) 0 :=
+{ hom := hom,
+  zero' := zero',
+  comm := by { intro i, rw [homological_complex.zero_f_apply, add_zero], refl, }, }
+
+/-- Homotopy to zero for maps constructed with `null_homotopic_map'` -/
+@[simps]
+def null_homotopy' (h : Π i j, c.rel j i → (C.X i ⟶ D.X j)) :
+  homotopy (null_homotopic_map' h) 0 := by
+{ apply null_homotopy (λ i j, dite (c.rel j i) (h i j) (λ _, 0)),
+  intros i j hij,
+  dsimp,
+  rw [dite_eq_right_iff],
+  intro hij',
+  exfalso,
+  exact hij hij', }
+
+/-! This lemma and the following ones can be used in order to compute
+the degreewise morphisms induced by the null homotopic maps constructed
+with `null_homotopic_map` or `null_homotopic_map'` -/
+@[simp]
+lemma null_homotopy_f {k₂ k₁ k₀ : ι} (r₂₁ : c.rel k₂ k₁) (r₁₀ : c.rel k₁ k₀)
+  (hom : Π i j, C.X i ⟶ D.X j) :
+  (null_homotopic_map hom).f k₁ = C.d k₁ k₀ ≫ hom k₀ k₁ + hom k₁ k₂ ≫ D.d k₂ k₁ := by
+{ dsimp [null_homotopic_map], rw [d_next_eq hom r₁₀, prev_d_eq hom r₂₁], }
+
+@[simp]
+lemma null_homotopy_f' {k₂ k₁ k₀  : ι} (r₂₁ : c.rel k₂ k₁) (r₁₀ : c.rel k₁ k₀)
+  (h : Π i j, c.rel j i → (C.X i ⟶ D.X j)) :
+  (null_homotopic_map' h).f k₁ = C.d k₁ k₀ ≫ h k₀ k₁ r₁₀ + h k₁ k₂ r₂₁ ≫ D.d k₂ k₁ := by
+{ simp only [← null_homotopic_map'],
+  rw null_homotopy_f r₂₁ r₁₀ (λ i j, dite (c.rel j i) (h i j) (λ _, 0)),
+  dsimp,
+  split_ifs,
+  refl, }
+
+@[simp]
+lemma null_homotopy_f_lower_end {k₁ k₀ : ι} (r₁₀ : c.rel k₁ k₀)
+  (hk₀ : ∀ l : ι, ¬c.rel k₀ l)
+  (hom : Π i j, C.X i ⟶ D.X j) :
+  (null_homotopic_map hom).f k₀ = hom k₀ k₁ ≫ D.d k₁ k₀ := by
+{ dsimp [null_homotopic_map],
+  rw prev_d_eq hom r₁₀,
+  rcases h : c.next k₀ with _|⟨l,w⟩, swap, exfalso, exact hk₀ l w,
+  dsimp [d_next], rw h, erw zero_add, }
+
+@[simp]
+lemma null_homotopy_f_lower_end' {k₁ k₀ : ι} (r₁₀ : c.rel k₁ k₀)
+  (hk₀ : ∀ l : ι, ¬c.rel k₀ l)
+  (h : Π i j, c.rel j i → (C.X i ⟶ D.X j)) :
+  (null_homotopic_map' h).f k₀ = h k₀ k₁ r₁₀ ≫ D.d k₁ k₀ := by
+{ simp only [← null_homotopic_map'],
+  rw null_homotopy_f_lower_end r₁₀ hk₀ (λ i j, dite (c.rel j i) (h i j) (λ _, 0)),
+  dsimp,
+  split_ifs,
+  refl, }
+
+@[simp]
+lemma null_homotopy_f_higher_end {k₁ k₀ : ι} (r₁₀ : c.rel k₁ k₀)
+  (hk₁ : ∀ l : ι, ¬c.rel l k₁)
+  (hom : Π i j, C.X i ⟶ D.X j) :
+  (null_homotopic_map hom).f k₁ = C.d k₁ k₀ ≫ hom k₀ k₁ := by
+{ dsimp [null_homotopic_map],
+  rw d_next_eq hom r₁₀,
+  rcases h : c.prev k₁ with _|⟨l,w⟩, swap, exfalso, exact hk₁ l w,
+  dsimp [prev_d], rw h, erw add_zero, }
+
+@[simp]
+lemma null_homotopy_f_higher_end' {k₁ k₀ : ι} (r₁₀ : c.rel k₁ k₀)
+  (hk₁ : ∀ l : ι, ¬c.rel l k₁)
+  (h : Π i j, c.rel j i → (C.X i ⟶ D.X j)) :
+  (null_homotopic_map' h).f k₁ = C.d k₁ k₀ ≫ h k₀ k₁ r₁₀ := by
+{ simp only [← null_homotopic_map'],
+  rw null_homotopy_f_higher_end r₁₀ hk₁ (λ i j, dite (c.rel j i) (h i j) (λ _, 0)),
+  dsimp,
+  split_ifs,
+  refl, }
+
+@[simp]
+lemma null_homotopy_f_middle_of_nowhere {k₀ : ι} 
```

Comment:
Here the `_middle_of_nowhere` could be replaced by `eq_zero`. The idea is that this accurately describes the statement of the lemma, and users can intuitively guess that there will be some side conditions required for this `= 0` to be true, even if they aren't described by the name.

Reply:
Thanks for the review. Indeed, I was not sure about the names... I have fixed these and the coding style also in the last commit.